### PR TITLE
OCS-284: Creating a pvc with an invalid name and size as inputs

### PR DESCRIPTION
**Pull request by PrasadDesala**
Signed-off-by: Prasad Desala <tdesala@redhat.com>

### CODE REVIEW

File: suites/manage/ocp/test_pvc_invalid_inputs.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    data['storage_class_name'] = storageclassname
+    data['blockpool_name'] = poolname
+    _templating = templating.Templating()
+    tmp_yaml_file = _templating.render_template(RBD_SC_YAML, data)
+
+    with open(TEMP_SC_YAML_FILE, 'w') as fd:
+        fd.write(tmp_yaml_file)
+        log.info(f"Creating RBD pool and storage class")
+    assert OCCLI.create(TEMP_SC_YAML_FILE)
+    log.info(f"RBD pool: {poolname} storage class: {storageclassname} "
```
Comment (RazTamir) [inline suites/manage/ocp/test_pvc_invalid_inputs.py]: fix brackets according to coding guidelines

File: suites/manage/ocp/test_pvc_invalid_inputs.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+             "created successfully")
+    log.info(TEMP_SC_YAML_FILE)
```
Comment (RazTamir) [inline suites/manage/ocp/test_pvc_invalid_inputs.py]: maybe in debug level

File: suites/manage/ocp/test_pvc_invalid_inputs.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    data['pvc_name'] = pvcname
+    _templating = templating.Templating()
+    tmp_yaml_file = _templating.render_template(RBD_PVC_YAML, data)
+    with open(TEMP_PVC_YAML_FILE, 'w') as fd:
+        fd.write(tmp_yaml_file)
+        log.info(f"Creating a pvc with name {pvcname}")
+    log.info(tmp_yaml_file)
+    oc_cmd = "oc "
+    kubeconfig = f"--kubeconfig {os.getenv('KUBECONFIG')}"
+    cmd = f"{oc_cmd} {kubeconfig} create -f {TEMP_PVC_YAML_FILE}"
```
Comment (RazTamir) [inline suites/manage/ocp/test_pvc_invalid_inputs.py]: why you are not using the existing create method?

File: suites/manage/ocp/test_pvc_invalid_inputs.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    try:
+        run_cmd(cmd)
+    except CommandFailed as ex:
+        if "error" in str(ex):
+            log.info(f"PVC creation failed with error \n {ex} \n as "
```
Comment (RazTamir) [inline suites/manage/ocp/test_pvc_invalid_inputs.py]: fix brackets

Comment (PrasadDesala) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: Done

File: suites/manage/ocp/test_pvc_invalid_inputs.py (hunk 2 above)
Comment (PrasadDesala) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: Done

File: suites/manage/ocp/test_pvc_invalid_inputs.py (hunk 1 above)
Comment (PrasadDesala) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: Done

File: suites/manage/ocp/test_pvc_invalid_inputs.py (hunk 3 above)
Comment (PrasadDesala) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: I will have to get the error of the oc command and do the validation from the error. 
I am able to achieve it through run_cmd() where as oc create is returning an munch object

Comment (RazTamir) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: You can use try and except to catch the exception in case create will fail

Comment (PrasadDesala) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: Done. Used OCP Create function.

File: suites/manage/ocp/test_pvc_invalid_inputs.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    # Create a storage class
+    log.info("Creating a Storage Class")
+    self.sc_data = defaults.STORAGECLASS_DICT.copy()
+    self.sc_data['metadata']['name'] = helpers.create_unique_resource_name(
+        'test', 'csi-rbd'
+    )
+    global SC_OBJ
+    global SC_NAME
+    SC_OBJ = OCS(**self.sc_data)
+    SC_NAME = self.sc_data['metadata']['name']
```
Comment (RazTamir) [inline suites/manage/ocp/test_pvc_invalid_inputs.py]: 2 things here:
1) this is a global parameter and therefore not need to return it as it is accessible everywhere in this file
2) why to keep it if you can access it everywhere with self.sc_data['metadata']['name']?

File: suites/manage/ocp/test_pvc_invalid_inputs.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    assert SC_OBJ.create()
+    log.info(f"Storage class: {SC_NAME} created successfully")
+    log.debug(self.sc_data)
+    return SC_NAME
```
Comment (RazTamir) [inline suites/manage/ocp/test_pvc_invalid_inputs.py]: setup and teardown usually not returning values

File: suites/manage/ocp/test_pvc_invalid_inputs.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    log.info(f"Deleting created storage class: {SC_NAME}")
+    SC_OBJ.delete()
+    log.info(f"Storage class: {SC_NAME} deleted successfully")
+
+
+@tier1
+@pytest.mark.usefixtures(test_fixture.__name__)
+class TestPvcCreationInvalidInputs(ManageTest):
+    """
+    This is the class for creating pvc with invalid inputs
```
Comment (RazTamir) [inline suites/manage/ocp/test_pvc_invalid_inputs.py]: please add the link to the Polarion TC or at least the Polarion ID

File: suites/manage/ocp/test_pvc_invalid_inputs.py [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    """
+    pvc_data = defaults.PVC_DICT.copy()
+    pvc_data['metadata']['name'] = pvcname
+    pvc_data['spec']['storageClassName'] = SC_NAME
+    PVC_OBJ = OCS(**pvc_data)
+    log.info(f"Creating a pvc with name {pvcname}")
+    try:
+        PVC_OBJ.create()
+    except CommandFailed as ex:
+        if "error" in str(ex):
```
Comment (RazTamir) [inline suites/manage/ocp/test_pvc_invalid_inputs.py]: you can be sure 'error' will be in ex.
Please search for something more meaningful

File: suites/manage/ocp/test_pvc_invalid_inputs.py [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    Args:
+        pvcname (str): Name of the pvc to be created
+
+    Returns:
+        None
+    """
+    pvc_data = defaults.PVC_DICT.copy()
+    pvc_data['metadata']['name'] = pvcname
+    pvc_data['spec']['storageClassName'] = SC_NAME
+    PVC_OBJ = OCS(**pvc_data)
```
Comment (RazTamir) [inline suites/manage/ocp/test_pvc_invalid_inputs.py]: please rebase your PR and use PVC class

File: suites/manage/ocp/test_pvc_invalid_inputs.py [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+        pvcsize (str): Size of the pvc to be created
+
+    Returns:
+        None
+    """
+    pvc_data = defaults.PVC_DICT.copy()
+    pvc_data['metadata']['name'] = "auto"
+    pvc_data['spec']['resources']['requests']['storage'] = pvcsize
+    pvc_data['spec']['storageClassName'] = SC_NAME
+    PVC_OBJ = OCS(**pvc_data)
```
Comment (RazTamir) [inline suites/manage/ocp/test_pvc_invalid_inputs.py]: use PVC class from pvc.py

File: suites/manage/ocp/test_pvc_invalid_inputs.py (hunk 6 above)
Comment (PrasadDesala) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: Done

File: suites/manage/ocp/test_pvc_invalid_inputs.py (hunk 7 above)
Comment (PrasadDesala) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: Done

File: suites/manage/ocp/test_pvc_invalid_inputs.py (hunk 8 above)
Comment (PrasadDesala) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: Done. Now validating with the entire error string.

File: suites/manage/ocp/test_pvc_invalid_inputs.py (hunk 9 above)
Comment (PrasadDesala) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: Done

File: suites/manage/ocp/test_pvc_invalid_inputs.py (hunk 10 above)
Comment (PrasadDesala) [reply suites/manage/ocp/test_pvc_invalid_inputs.py]: Done

**Comment by PrasadDesala** [comment]
Verified:
========================================================================== short test summary info ==========================================================================
PASSED test_pvc_invalid_inputs.py::TestPvcCreationInvalidInputs::test_pvccreation_invalid_inputs
========================================================================= 1 passed in 9.20 seco

**Pull request merged**